packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -209,11 +211,25 @@ export default class Proxy {
       xoaUpgradeTimeout
     )
 
-    const { id } = await this.registerProxy({
+    const props = {
       authenticationToken: proxyAuthenticationToken,
       name: this._generateDefaultProxyName(date),
       vmUuid: vm.uuid,
-    })
+    }
+
+    let id = proxyId
+    if (id !== undefined) {
+      const proxy = await this.getProxy(proxyId)
+      if (proxy.vmUuid !== undefined) {
```

Comment:
When? proxy should be deployed, so it should have a `vmUuid`. no?

Reply:
The functionality is about re-deploying the same proxy. A `VM` can be unliked from a proxy by the user.